### PR TITLE
feat : add subscription functionality

### PR DESCRIPTION
**Pull request by cp-mansi-d**
Google Authentication with Firebase for Flutter Desktop Apps _ by _ published at May 16, 2023 _ Cano

https://github.com/canopas/canopas-blog/assets/115449373/1d4ca870-e7f3-418c-8cd2-f1f81c430296

## Review checklist

**Note:** Make sure all points should be checked before submiting PR for review, otherwise PR will be cosidered invalid.

- [x] Follow guidelines for creating branch name, adding commit messages or PR title
- [x] Add MR description if applicable
- [x] Follow guidelines for naming conventions everywhere (i.e files/folders, data structures)
- [x] Reuse code (if the same code is written twice, make it common and reuse it at both places)
- [x] Remove unused or commented code if not required
- [x] Make MR self-approved - review MR as a reviewer and give it self-approval if everything is ok, if not make the required changes
- [x] In progress MR should be marked as draft
- [x] Make MR mark as ready before submitting it for review
- [x] Add images, videos, or gifs of UI
- [ ] Add page speed score screenshot of UI (It should be > 85 for mobile and > 90 for desktops)

**Design should be tested on:** - **testing for image loading problem**

- [x] Inspector (all iPhones and android devices like samsung, motorola)
- [x] Desktop (chrome, firefox, safari)
- [ ] Laptop / Macbook (chrome, firefox, safari)
- [x] iPhone(safari)
- [x] Available android devices (chrome, firefox)
- [x] iPad(safari)
- [x] Tablet (chrome, firefox)

**Description**
- added subscription functionality for user. it provide to subscribe or unsubscribe for users.
- and fixed some image loading problem in first time loading and hard refresh. 
- And I also added the name field and removed the username field in the user table, so I changed some code in the comment section e.g. When the user adds their name it returns the user id and creates a comment if the name already exists at that time, but when the user adds a name and that name does not exist then update that username in the user table and the username must also be unique.

### CODE REVIEW

File: pages/[slug].js [hunk 1]
```diff
@@ -466,25 +487,61 @@ export default function Post({ postData, status, categoryPosts, mixpanel }) {
...
+
+                    <button className="w-auto rounded-full border-[1px] border-solid border-transparent bg-gradient-to-r from-[#f2709c] to-[#ff9472] text-[1.1rem] font-semibold text-white hover:shadow-[inset_2px_1000px_1px_#fff] hover:cursor-pointer">
+                      <div className="px-[1.35rem] py-[0.5rem] align-middle text-center tracking-wide gradient-text">
+                        Subscribe
+                      </div>
+                    </button>
+                  </form>
+
+                  <div className="justify-self-center md:justify-self-end w-40 md:w-8/12 xl:w-7/12">
+                    <Image
```
Comment (cp-sumi-k) [inline pages/[slug].js:514]: Remove this icon from mobile devices

File: pages/[slug].js [hunk 2]
```diff
@@ -126,6 +129,23 @@ export default function Post({ postData, status, categoryPosts, mixpanel }) {
...
+    event.preventDefault();
+
+    await axios
+      .post(`${config.STRAPI_URL}/v1/user/subscribeUser?populate=deep`, {
+        email,
+      })
+      .then((response) => {
+        setCopied(true);
+        if (response.data.is_subscribed) {
+          setMessage("Subscribe Successfully!");
```
Comment (cp-sumi-k) [inline pages/[slug].js:142]: Show this message at the top of the page, where we are showing the `Link copied` message. Or you can use sam prompt, by changes messages

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: pages/[slug].js (hunk 2 above)
Comment (cp-mansi-d) [reply pages/[slug].js:142]: we are already display `link copied` message at bottom of the page. and i used same alerts ui for messages print,only chnage message.

File: pages/[slug].js [hunk 3]
```diff
@@ -148,7 +172,8 @@ export default function Post({ postData, status, categoryPosts, mixpanel }) {
     el.select();
     document.execCommand("copy");
     document.body.removeChild(el);
-    setCopied(true);
+    setAlerts(true);
+    setMessage("Link alerts");
```
Comment (cp-sumi-k) [inline pages/[slug].js:176]: I think it should be `link copied`

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**